timex/timex.go

```diff
@@ -0,0 +1,57 @@
+// Package timex should be called Rolex.
+package timex
+
+import (
+	"context"
+	"time"
+)
+
+// Ticker ...
+type Ticker struct {
+	C <-chan time.Time
+
+	tt   *time.Ticker
+	done chan bool
+	c    chan time.Time
+}
+
+// NewTicker ...
+func NewTicker(d time.Duration) *Ticker {
+	c := make(chan time.Time, 1)
+	ticker := Ticker{
+		C:    c,
+		tt:   time.NewTicker(d),
+		done: make(chan bool),
+		c:    c,
+	}
+
+	go func() {
+		for {
+			select {
+			case <-ticker.done:
+				return
+			case t := <-ticker.tt.C:
+				ticker.c <- t
+			}
+		}
+	}()
+
+	return &ticker
+}
+
+// Stop ...
+func (t *Ticker) Stop() {
+	t.tt.Stop()
+	t.done <- true
+	close(t.c)
```

Comment:
If you closed the `done` channel here instead, you could get rid of the `c` field. It would also be safer as sending to the done channel could block if Stop is called more than once. To close the channel only once (otherwise it panics) you can do:
```
select {
  case <-t.done:
  default: close(t.done)
}
```
The default case should only fire if `t.done` is not ready, and `t.done` will always be ready if it is closed.

Reply:
I deleted part of this comment by accident. Making this close `t.done`, you move the `close(c)` into the select case on t.done in the goroutine.

Reply:
hmm need to think about this more tomorrow. done.

Reply:
I have a sample implementation at https://goplay.space/#8tdxdbh2olT which pulls together all my comments here.

Reply:
Nice, well update shortly.